src/app/problems/problem-view/problem-view.component.ts

```diff
@@ -41,4 +50,24 @@ export class ProblemViewComponent implements OnInit {
             this.user = user;
         });
     }
+
+    switchFavorite(uqjId: number, favoriteStatus: boolean, questionId: number): void {
+        const data: { id: number, status: boolean } = {id: uqjId, status: !favoriteStatus};
```

Comment:
same as before. Make it a data type if you think it will be used in multiple places